src/Calendar.jsx

```diff
@@ -153,7 +157,7 @@ let Calendar = React.createClass({
     //if the value changes reset views to the new one
     if ( !dates.eq(val, dateOrNull(this.props.value), VIEW_UNIT[view]))
       this.setState({
-        currentDate: val ? new Date(val) : new Date()
+        currentDate: this.getDateOrCurrent(val)
```

Comment:
So the way uncontrollable works is that the handler/prop pair are managed for you like they might be if you `setState()`

In this case you have `onCurrentDateChange` and `currentDate` when you want to "set" current date, instead of setting the state, pass that value to `this.props.onCurrentDateChange`. You can look at how `value` and `onChange` work here.